homeassistant/components/sense/binary_sensor.py

```diff
@@ -49,17 +49,15 @@
 }
 
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
     """Set up the Sense binary sensor."""
-    if discovery_info is None:
```

Comment:
We should keep this guard clause. It's needed to only act on setup via discovery.